src/provider/index.ts

```diff
@@ -40,6 +42,8 @@ export class Main {
     @inject(Inject.DATABASE)
     private _database!: Database;
 
+    private _centerBlurTimerHandle: number | undefined;
```

Comment:
Agree that the `Debounced` util as it exists in Layouts doesn't really fit here, but a "throttled" util?.. Something that's the same thing, but triggers the callback on the "rising edge" rather than the "falling edge"? Given the similarities, could maybe be the same util with some argument?

With a throttled approach, I don't think we'd need the additional `BLUR` action, etc. Rather calls to `toggleVisibility` (or whatever we decide the common entry point is) would have no effect if called within some short duration after a previous call.

Reply:
I disagree that we want to get rid of the distinction between Blur and Toggle. It's only one of those we want to apply this filtering to. And whether it makes sense to pull this filtering code into its own util is an orthogonal question

Reply:
I tried pulling out ` `Throttled` class, that worked vaguely analogously to `Debounced`, but it wasn't a lot of use with things as they are, so it's ended up as just a `Timer` class.

Were you suggesting move the throttling logic down into `Actions.ts`? The timer/throttler/whatever would need to live at the module level so it could be referenced by `setCenterVisibility`, which feels wrong?

Reply:
Using `ToggleFilter`